doc/wiki/dashboard-audit.md

```diff
@@ -0,0 +1,668 @@
+
+---
+
+## Visited but no BP taken
+**Copy used in the Dashboard**
++ Numerator: Patients with no BP taken at their last visit in the last 3 months
++ Denominator: Hypertension patients assigned to `region_name`, registered before the last 3 months. Dead and lost to follow-up patients are exclued.
+
+**How it's calculated**
+The number of patients assigned to a facility, registered before the last 3 moths where the patient:
++ Is not deleted
++ Is hypertensive
++ Is not dead
++ Has at least one of the following: An appointment created, A drug refilled, A blood sugar taken
++ Doesn't have a BP recorded within the last 3 months
+
+*Sample SQL query*
+```sql
+SELECT COUNT(DISTINCT "patients"."id")
+FROM "patients"
+INNER JOIN "medical_histories" ON "medical_histories"."deleted_at" IS NULL
+AND "medical_histories"."patient_id" = "patients"."id"
+LEFT OUTER JOIN latest_blood_pressures_per_patient_per_months ON patients.id = latest_blood_pressures_per_patient_per_months.patient_id
+LEFT OUTER JOIN appointments ON appointments.patient_id = patients.id
+AND appointments.device_created_at >= '2019-02-28 18:30:00'
+AND appointments.device_created_at <= '2019-05-31 18:29:59.999999'
+LEFT OUTER JOIN prescription_drugs ON prescription_drugs.patient_id = patients.id
+AND prescription_drugs.device_created_at >= '2019-02-28 18:30:00'
+AND prescription_drugs.device_created_at <= '2019-05-31 18:29:59.999999'
+LEFT OUTER JOIN blood_sugars ON blood_sugars.patient_id = patients.id
+AND blood_sugars.recorded_at >= '2019-02-28 18:30:00'
+AND blood_sugars.recorded_at <= '2019-05-31 18:29:59.999999'
+WHERE "patients"."deleted_at" IS NULL
+  AND "medical_histories"."deleted_at" IS NULL
+  AND "medical_histories"."hypertension" = $1
+  AND "patients"."status" != $2
+  AND (bp_recorded_at > '2018-05-31'
+       AND bp_recorded_at < '2019-05-31 18:29:59.999999'
+       OR patients.recorded_at >= '2018-05-31')
+  AND "patients"."assigned_facility_id" IN $3
+  AND (patients.recorded_at <= '2019-02-28 18:30:00')
+  AND (appointments.id IS NOT NULL
+       OR prescription_drugs.id IS NOT NULL
+       OR blood_sugars.id IS NOT NULL)
+  AND (NOT EXISTS
+         (SELECT 1
+          FROM blood_pressures bps
+          WHERE patients.id = bps.patient_id
+            AND bps.recorded_at >= '2019-02-28 18:30:00'
+            AND bps.recorded_at <= '2019-05-31 18:29:59.999999')) [["hypertension", "yes"],
+                                                                   ["status", "dead"],
+                                                                   ["assigned_facility_id", "3a7e86d2-c272-4303-8ffa-d6d1b54874b3"]]
+```
+
+**Where it's shown in the Dashboard**
++ **Reports:** Overview
+
+**Things we should fix**
++ [ ] Review LTFU code paths in `Result` and `Repository` objects
```

Comment:
What are we reviewing them for? LTFU is very new, so I would expect it to be accurate. Is the issue that something might have been missed when Rob added Repository?

Reply:
@prabhanshuguptagit and I have reviewed this changes pretty closely as we have both been working on these code paths.  So I don't think we need a review here, as its already happened. =)